sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -321,6 +326,7 @@ internal DataLakePathClient(Uri pathUri, HttpPipelinePolicy authentication, Data
             _version = options.Version;
             _clientDiagnostics = new ClientDiagnostics(options);
             _blockBlobClient = BlockBlobClientInternals.Create(_blobUri, _pipeline, Version.AsBlobsVersion(), _clientDiagnostics);
+            _options = options;
```

Comment:
This is kind of problematic.  The pipeline is designed to be immutable but the client options are not.  We're supposed to cache any values needed from the options when creating the pipeline rather than hanging on to the options object directly.  We already do that for things like the diagnostics, version, etc.  Is there any reason we can't keep doing that?

Reply:
Fixed, we didn't need this.